zebra-state/src/service/non_finalized_state/tests/prop.rs

```diff
@@ -14,19 +15,20 @@ fn forked_equals_pushed() -> Result<()> {
                                           .ok()
                                           .and_then(|v| v.parse().ok())
                                           .unwrap_or(DEFAULT_PARTIAL_CHAIN_PROPTEST_CASES)),
-        |((chain, count, _network) in PreparedChain::default())| {
+        |((chain, count, network) in PreparedChain::default())| {
+            let dummy_tree = HistoryTree::new_from_block(network, chain[0].block.clone(), &sapling::tree::Root::default(), None).unwrap();
             let fork_tip_hash = chain[count - 1].hash;
-            let mut full_chain = Chain::default();
-            let mut partial_chain = Chain::default();
+            let mut full_chain = Chain::new(dummy_tree.clone());
+            let mut partial_chain = Chain::new(dummy_tree.clone());
 
             for block in chain.iter().take(count) {
-                partial_chain.push(block.clone());
+                partial_chain.push(block.clone())?;
             }
             for block in chain.iter() {
-                full_chain.push(block.clone());
+                full_chain.push(block.clone())?;
             }
 
-            let forked = full_chain.fork(fork_tip_hash).expect("hash is present");
+            let forked = full_chain.fork(fork_tip_hash, &dummy_tree).expect("fork must work").expect("hash is present");
 
             prop_assert_eq!(forked.blocks.len(), partial_chain.blocks.len());
```

Comment:
Testing: The current tests are a good start, but let's add some steps to this test:
1. Make sure `forked` and `partial_chain` are identical
  - the `PartialEq` impl only checks work and block hash, so let's add a test-only `is_identical` method
2. Push the remaining blocks onto `forked`
3. Make sure the chains match